rockcraft.yaml

```diff
@@ -0,0 +1,170 @@
+    source: https://github.com/CloudNativeDataPlane/cndp.git
+    source-commit: d5ce4b9edc2e7ddb46a61b395deffafaf11a0500
+    build-environment:
+      - CNE_DEST_DIR: "${CRAFT_PART_INSTALL}"
+    build-packages:
+      - clang
+      - golang
+      - libbpf-dev
+      - libbsd-dev
+      - libnl-3-dev
+      - libnl-cli-3-dev
+      - libnuma-dev
+      - lld
+    override-build: |
+      make
+      make install
+      make static_build=1 rebuild install
+      pushd lang/go/stats/prometheus/
+      go build -o $CRAFT_PART_INSTALL/prometheus prometheus.go
+    prime:
+      - usr/local/lib/x86_64-linux-gnu/*.so
+      - usr/local/bin/cndpfwd
+
+  dpdk:
+    plugin: meson
+    after:
+      - cndp
+    source: https://fast.dpdk.org/rel/dpdk-20.11.3.tar.gz
+    source-type: tar
+    build-packages:
+      - ca-certificates
+      - libbenchmark-dev
+      - libbpf0
+      - libc-ares-dev
+      - libelf-dev
+      - libgflags2.2
+      - libgoogle-glog-dev
+      - libgrpc++-dev
+      - libgtest-dev
+      - libjson-c-dev
+      - libnuma-dev
+      - libpcap-dev
+      - libprotobuf-dev
+      - libunwind-dev
+      - meson
+      - protobuf-compiler
+      - protobuf-compiler-grpc
+      - python3
+    override-pull: |
+      craftctl default
+      for file in ${CRAFT_PROJECT_DIR}/local/patches/*
+      do
+        patch -i $file -p 1
+      done
+    meson-parameters:
+      - --buildtype=debugoptimized
+      - -Dmachine=x86-64
+    prime:
+      - usr/local/lib/x86_64-linux-gnu/*.so
+
+  bess:
+    plugin: nil
+    after:
+      - dpdk
+    source: https://github.com/omec-project/bess.git
+    source-tag: dpdk-2011-focal
+    source-type: git
+    build-packages:
+      - ca-certificates
+      - libbenchmark-dev
+      - libbpf0
+      - libc-ares-dev
+      - libelf-dev
+      - libgflags2.2
+      - libgoogle-glog-dev
+      - libgrpc++-dev
+      - libgtest-dev
+      - libjson-c-dev
+      - libnuma-dev
+      - libpcap-dev
+      - libprotobuf-dev
+      - libssl-dev
+      - libunwind-dev
+      - meson
+      - pkg-config
+      - protobuf-compiler
+      - protobuf-compiler-grpc
+      - python3
+    stage-packages:
+      - libbpf0
+      - libgflags2.2
+      - libjson-c4
+      - libgraph-easy-perl
+      - iproute2
+      - iptables
+      - iputils-ping
+      - tcpdump
+      - ethtool
+    override-build: |
+      for file in $(find ./protobuf -name *.proto -print)
+      do
+        protoc "$file" --proto_path=protobuf --python_out=pybess/builtin_pb --grpc_out=pybess/builtin_pb --plugin=protoc-gen-grpc=$(which grpc_python_plugin)
+      done
+      make -C core bessd modules
+      mkdir -p $CRAFT_PART_INSTALL/bin
+      cp bin/bessd $CRAFT_PART_INSTALL/bin/bessd
+      mkdir -p $CRAFT_PART_INSTALL/opt/bess/bessctl
+      mkdir -p $CRAFT_PART_INSTALL/opt/bess/pybess
+      cp -r bessctl/* $CRAFT_PART_INSTALL/opt/bess/bessctl/
+      cp -r pybess/* $CRAFT_PART_INSTALL/opt/bess/pybess/
+
+  bess-python-deps:
```

Comment:
Should we consider leaving aside the routectl considerations in this first version of the ROCK?

Reply:
Since `routectl` is just a small addition on top of `bess` code, and that `routectl` will always run in the same pod as `bessd`, I think keeping both in the same image is reasonable and will probably save some download time.

Reply:
What does routectl depend on? Is it only the code in the conf directory?

Reply:
It also depends on the `pybess` module, from the `bess` project.